frontend/src/pages/Register/index.jsx

```diff
@@ -48,14 +77,36 @@ const Register = () => {
   const userSelector = useSelector((state) => state.user);
 
   // State
+  const municipalities = useGetValidMunicipalities();
+  const domains = municipalities ? municipalities.map((mun) => mun.mailDomain) : [];
   const [email, setEmail] = useState('');
+  const [type, setType] = useState(1);
   const [password, setPassword] = useState('');
   const [verifyPassword, setVerifyPassword] = useState('');
   const [loading, setLoading] = useState(false);
 
   // Redux
   const dispatch = useDispatch();
 
+  const validMunicipalityEmail = (_email) => {
+    for (const domain of domains) { // eslint-disable-line no-restricted-syntax
+      if (_email.endsWith(domain)) {
```

Comment:
the email should end with `` `@${domain}` ``, not just domain. Currently this accepts e.g. `alex@basker.kommune.no`, since this ends on `asker.kommune.no`